packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -1056,20 +1061,21 @@ function updateClassInstance(
     // effect even though we're bailing out, so that cWU/cDU are called.
     if (typeof instance.componentDidUpdate === 'function') {
       if (
-        oldProps !== current.memoizedProps ||
+        unresolvedOldProps !== current.memoizedProps ||
         oldState !== current.memoizedState
       ) {
         workInProgress.effectTag |= Update;
       }
     }
     if (typeof instance.getSnapshotBeforeUpdate === 'function') {
       if (
-        oldProps !== current.memoizedProps ||
+        unresolvedOldProps !== current.memoizedProps ||
         oldState !== current.memoizedState
       ) {
         workInProgress.effectTag |= Snapshot;
       }
     }
+    instance.props = workInProgress.memoizedProps = newProps;
```

Comment:
Why is this needed?

Reply:
Otherwise, React would warn.
https://github.com/facebook/react/blob/1fd45437d781b1b905be2308447aa7793bbcb571/packages/react-reconciler/src/ReactFiberBeginWork.js#L859-L868

This also mirrors the logic at the end of the same function.
https://github.com/facebook/react/blob/1fd45437d781b1b905be2308447aa7793bbcb571/packages/react-reconciler/src/ReactFiberClassComponent.js#L1139-L1151

This was not needed before because it was already guaranteed by the condition in the branch.
https://github.com/facebook/react/blob/1fd45437d781b1b905be2308447aa7793bbcb571/packages/react-reconciler/src/ReactFiberClassComponent.js#L1050